cloudinit/sources/DataSourceIBMCloud.py

```diff
@@ -325,27 +334,39 @@ def read_md():
     return ret
 
 
-def metadata_from_dir(source_dir):
+def metadata_from_dir(source_dir: str) -> Dict[str, Any]:
     """Walk source_dir extracting standardized metadata.
 
     Certain metadata keys are renamed to present a standardized set of metadata
     keys.
 
     This function has a lot in common with ConfigDriveReader.read_v2 but
-    there are a number of inconsistencies, such key renames and as only
-    presenting a 'latest' version which make it an unlikely candidate to share
+    there are a number of inconsistencies, such as key renames and only 
+    presenting a 'latest' version, which make it an unlikely candidate to share
     code.
 
     @return: Dict containing translated metadata, userdata, vendordata,
         networkdata as present.
     """
 
-    def opath(fname):
+    def opath(fname: str) -> str:
         return os.path.join("openstack", "latest", fname)
 
-    def load_json_bytes(blob):
+    def load_json_bytes(blob: bytes) -> Dict[str, Any]:
```

Comment:
sorry I'm somewhat unclear what you're referring to. are you suggesting the following?
```suggestion
    def load_json_bytes(blob: bytes) -> Dict[str, Union[List,str]]:
```

Reply:
The return type is specified as `Dict[...]`, but `json.loads` can return `List` and `str` too. Changing it here would likely will make the code that calls this more annoying, and I think it's ok to assume we're going to get a dict. We may just want to add a comment that this may not be technically correct.